### PR TITLE
make classes static where possible

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/NUnitFramework/framework/Assume.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace NUnit.Framework
     /// result. If an assumption is not met, the test
     /// should produce an inconclusive result.
     /// </summary>
-    public class Assume
+    public static class Assume
```
Comment (jnm2) [inline src/NUnitFramework/framework/Assume.cs:19]: We shouldn't make public classes outside the Internal namespace static. A publicized technique for "extending" the NUnit assertion classes is to do this:

```cs
public class Assert : NUnit.Framework.Assert
{
    public static void Foo(...) { }
}
```

Comment (jnm2) [reply src/NUnitFramework/framework/Assume.cs:19]: We could consider making them `abstract` to break code which does `new Assert()` and I thought we had done so.

**Review by jnm2 — CHANGES_REQUESTED** [review]

File: src/NUnitFramework/framework/Assume.cs (hunk 1 above)
Comment (SimonCropp) [reply src/NUnitFramework/framework/Assume.cs:19]: so revert the Assume and Warn changes?

File: src/NUnitFramework/framework/Assume.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace NUnit.Framework
     /// result. If an assumption is not met, the test
     /// should produce an inconclusive result.
     /// </summary>
-    public class Assume
+    public abstract class Assume
```
Comment (rprouse) [inline src/NUnitFramework/framework/Assume.cs:19]: Why `abstract` and not `static`? I don't see anything that makes this `abstract` and it is used as-is? Same with the other usages of `abstract`.

Comment (rprouse) [reply src/NUnitFramework/framework/Assume.cs:19]: Nevermind, I see Joseph's comment. That is a good point.

**Review by rprouse — APPROVED** [review]

**Pull request merged**